scripts/globals/follow.lua

```diff
@@ -56,30 +57,59 @@ function addFollower(leader, follower, options)
     local leaderId = leader:getID()
     local followerId = follower:getID()
 
-    if followerToLeaderMap[followerId] then
-        removeFollower(followerToLeaderMap[followerId], follower)
-    end
+    xi.follow.stopFollowing(follower)
 
     leaderToFollowersMap[leaderId] = leaderToFollowersMap[leaderId] or {}
 
     table.insert(leaderToFollowersMap[leaderId], follower)
     followerToLeaderMap[followerId] = leader
 
     follower:addListener("ROAM_ACTION", "FOLLOW_ROAM_ACTION", onMobRoamAction)
+    follower:addListener("DESPAWN", "FOLLOW_DESPAWN", onMobDespawn)
+    follower:setLocalVar("[follow]roamFlags", follower:getRoamFlags())
+    follower:setLocalVar("[follow]DONT_ROAM_HOME", follower:getMobMod(xi.mobMod.DONT_ROAM_HOME))
+    follower:setLocalVar("[follow]ROAM_COOL", follower:getMobMod(xi.mobMod.ROAM_COOL))
     follower:setRoamFlags(xi.roamFlag.SCRIPTED)
     follower:setMobMod(xi.mobMod.DONT_ROAM_HOME, 1)
-    follower:setMobMod(xi.mobMod.ROAM_COOL, options.roamCooldown)
+    if options.roamCooldown then
+        follower:setMobMod(xi.mobMod.ROAM_COOL, options.roamCooldown)
+    end
+
     followerOptions[followerId] = options
 end
 
-function removeFollower(leader, follower)
-    local leaderId = leader:getID()
+function cleanupFollower(follower)
```

Comment:
Since this is a global function, can it be changed to `xi.follow.cleanupFollower`?

Reply:
`cleanupFollower()` is a local function. I declared it local at the beginning of the file. That said, it's only called in one place (as a result of refactoring from when I added it) so I'll just roll it into `xi.follow.stopFollowing()`

Reply:
Didn't catch it previously (way back in [august last year](https://github.com/LandSandBoat/server/commit/cb3458a2553f54546b3d2f52e8f53e2fbb406722#diff-840a590675fe79f6b29c82d4bb7eed93d24c0c273a5ea5a5d397aed8690670c9R18)) but I dislike the pattern to functions beginning their lives as nil variables. Which is why nobody noticed this was local and not global.

Reply:
For clarity, it's a way to keep functions in local scope but have the function definition after the function definition of a function that will call the local one. Since Lua is a single-pass compiler. I'm still using it for the `onMobDespawn` and `onMobRoamAction` callbacks.

If it's an approach that LSB would prefer to avoid, let me know.

Reply:
Definitely want to avoid.

Reply:
Okay, I've re-ordered the functions, as requested.